packages/tpetra/core/test/CrsMatrix/CrsMatrix_PackUnpack.cpp

```diff
@@ -539,9 +542,214 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(CrsMatrix, PackError, SC, LO, GO, NT)
   }
 }
 
+// PackPartial exercises the use case where only *some* of the rows of the
+// matrix are packed.  This test exists because an errant check in pack/unpack
+// checked that the number of rows to pack was equal to the number of rows in
+// the matrix.  However, this is not a requirement, and that particular check
+// caused existing code to fail.  See Issues #1374 and #1408
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(CrsMatrix, PackPartial, SC, LO, GO, NT)
+{
+  using std::endl;
+  using Tpetra::CrsMatrix;
+  using Tpetra::Details::gathervPrint;
+  using Tpetra::Details::packCrsMatrix;
+  using Tpetra::Details::unpackCrsMatrixAndCombine;
+  using Teuchos::Array;
+  using Teuchos::ArrayView;
+  using Teuchos::Comm;
+  using Teuchos::outArg;
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+  using Teuchos::REDUCE_MIN;
+  using Teuchos::REDUCE_SUM;
+  using Teuchos::reduceAll;
+  typedef CrsMatrix<SC, LO, GO, NT, false> crs_matrix_type;
+  typedef typename NT::device_type device_type;
+  typedef typename device_type::execution_space execution_space;
+
+  int lclSuccess = 1; // to be revised below
+  int gblSuccess = 0; // output argument
+
+  RCP<const Comm<int> > comm =
+    Tpetra::DefaultPlatform::getDefaultPlatform ().getComm ();
+  const int world_size = comm->getSize();
+  const int world_rank = comm->getRank();
+
+  // MPI_Comm_size returns the same value on all processes in the
+  // communicator, so we don't need to do a test before exiting early.
+  TEST_ASSERT( world_size % 2 == 0 );
+  if (world_size % 2 != 0) {
+    out << "This test must be run with multiples of 2 MPI processes, but was "
```

Comment:
It doesn't look like this test depends on MPI at all.  You're just packing data and unpacking it.  The test should work just fine with any number >= 1 of MPI processes.